src/shacl-shape-lib/elements-shacl-shape.xsl

```diff
@@ -17,102 +17,134 @@
+                <xsl:with-param name="className" select="$className"/>
             </xsl:call-template>
             <xsl:call-template name="attributeMultiplicity">
                 <xsl:with-param name="attribute" select="."/>
+                <xsl:with-param name="className" select="$className"/>
             </xsl:call-template>
         </xsl:if>
-    </xsl:template>-->
-
-
-    <xd:doc>
-        <xd:desc>[Rule 30]-(Label) Specify a label for UML element</xd:desc>
-        <xd:param name="name"/>
-    </xd:doc>
-    <xsl:template name="elementName">
-        <xsl:param name="name"/>
-        <sh:name>
-            <xsl:value-of select="$name"/>
-        </sh:name>
-<!--        <rdfs:label xml:lang="en">
-            <xsl:value-of select="$name"/>
-        </rdfs:label>-->
     </xsl:template>
 
+
     <xd:doc>
-        <xd:desc>[Rule 31]-(Description) Specify a description for UML element.</xd:desc>
-        <xd:param name="description"/>
+        <xd:desc>Rule C.06. Class attribute — in data shape layer. 
+            Specify a SHACL PropertyShape declaration axiom for each attribute.</xd:desc>
+       
+        <xd:param name="attribute"/>
+        <xd:param name="className"/>
     </xd:doc>
-    <xsl:template name="elementDescription">
-        <xsl:param name="description"/>
-        <sh:description>
-            <xsl:value-of select="$description"/>
-        </sh:description>
-<!--        <rdfs:comment xml:lang="en">
-            <xsl:value-of select="$description"/>
-        </rdfs:comment>-->
+    
+    <xsl:template name="classAttributeDeclaration">
+        <xsl:param name="attribute"/>
+        <xsl:param name="className"/>
+        <xsl:variable name="attributeURI"
+            select="f:buildURIFromElement($attribute)"/>
+        <xsl:variable name="documentation" select="$attribute/documentation/@value"/>
+        <xsl:variable name="normalisedAttributeName" select="replace($attribute/@name, ':', '-')"/>
+        <xsl:variable name="normalisedClassName" select="replace($className, ':', '-')"/>
+        <xsl:variable name="shapePropertyUri"
+            select="fn:concat($base-shape-uri, $defaultDelimiter, $normalisedClassName, '-', $normalisedAttributeName)"/>
+        <xsl:variable name="shapeClassUri"
+            select="fn:concat($base-shape-uri, $defaultDelimiter, replace($className, ':', '-'))"/>
+        
+        <rdf:Description rdf:about="{$shapeClassUri}">
+            <sh:property rdf:resource="{$shapePropertyUri}"/>
+        </rdf:Description>
+        <sh:PropertyShape rdf:about="{$shapePropertyUri}">
+            <sh:path rdf:resource="{$attributeURI}"/>
+        </sh:PropertyShape>
+        
+        <xsl:call-template name="shapeLayerName">
+            <xsl:with-param name="elementName" select="$attribute/@name"/>
+            <xsl:with-param name="uri" select="$shapePropertyUri"/>
+        </xsl:call-template>
+        <xsl:if test="$documentation != ''">
```

Comment:
YES, this is a professional piece of code. You broke it down into smaller triples and the assembled the templates. 

It is not completely correct here, however, because the descriptors shall have their own rules.